ui.frontend/src/HTTPAPILayer.js

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+import {Constants} from "./constants";
+
+export default class HTTPAPILayer {
+
+    static async getForm(formContainerPath) {
+        const _formPath = formContainerPath.substring(0, formContainerPath.indexOf("/jcr:content/guideContainer"));
+        const _formsList =  await this.#getFormsList();
+        if (_formsList) {
+            return await this.#findForm(_formPath, _formsList);
+        } else {
+            //TODO: throw errors once API is available on Circle CI set up
+            console.debug("Error in fetching form");
+        }
+    }
+
+    static async getFormDefinition(formContainerPath) {
+        return await this.#getJson(formContainerPath + ".model.json");
+    }
+
```

Comment:
I am still not comfortable with making so many network calls to get the form definition.

Reply:
there's still only one call required to fetch the model definition.
but for prefill call, we need form id, and hence, we will need listForms call.
issue is already logged to optimise it. 
https://github.com/adobe/aem-core-forms-components/issues/190